compiler/rustc_infer/src/infer/canonical/mod.rs

```diff
@@ -63,8 +63,8 @@ impl<'cx, 'tcx> InferCtxt<'cx, 'tcx> {
         // in them, so this code has no effect, but it is looking
         // forward to the day when we *do* want to carry universes
         // through into queries.
-        let universes: IndexVec<ty::UniverseIndex, _> = std::iter::once(ty::UniverseIndex::ROOT)
```

Comment:
Currently reading about Universes/Placeholders and am curious, why can we do that? Why can't we encounter canonicalized variables with `UniverseIndex::Root` inside `instantiate_canonical_vars`?